sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py

```diff
@@ -546,85 +546,74 @@ def request_id(self):
-        reuse_key=None,  # type: Optional[bool]
-        curve=None,  # type: Optional[KeyCurveName]
-        ekus=None,  # type: Optional[list[str]]
-        key_usage=None,  # type: Optional[list[Union[KeyUsageType, str]]]
-        content_type=None,  # type: Optional[Union[SecretContentType, str]]
-        validity_in_months=None,  # type: Optional[int]
-        lifetime_actions=None,  # type: Optional[list[LifetimeAction]]
-        certificate_type=None,  # type: Optional[str]
-        certificate_transparency=None,  # type: Optional[bool]
         **kwargs  # type: **Any
     ):
         # type: (...) -> None
-        self._subject_name = subject_name
-        self._attributes = kwargs.get("attributes", None)
-        self._id = kwargs.get("cert_policy_id", None)
-        self._exportable = exportable
-        self._key_type = key_type
-        self._key_size = key_size
-        self._reuse_key = reuse_key
-        self._curve = curve
-        self._ekus = ekus
-        self._key_usage = key_usage
-        self._content_type = content_type
-        self._validity_in_months = validity_in_months
-        self._lifetime_actions = lifetime_actions
         self._issuer_name = issuer_name
-        self._certificate_type = certificate_type
-        self._certificate_transparency = certificate_transparency
-        self._san_emails = kwargs.pop("san_emails", None)
-        self._san_dns_names = kwargs.pop("san_dns_names", None)
-        self._san_upns = kwargs.pop("san_upns", None)
-
-        sans = [self._san_emails, self._san_upns, self._san_dns_names]
-        if len([x for x in sans if x is not None]) > 1:
-            raise ValueError("You can only set at most one of san_emails, san_dns_names, and san_upns")
+        self._subject_name = kwargs.pop("subject_name", None)
+        self._attributes = kwargs.pop("attributes", None)
+        self._id = kwargs.pop("cert_policy_id", None)
+        self._exportable = kwargs.pop("exportable", None)
+        self._key_type = kwargs.pop("key_type", None)
+        self._key_size = kwargs.pop("key_size", None)
+        self._reuse_key = kwargs.pop("reuse_key", None)
+        self._curve = kwargs.pop("curve", None)
+        self._ekus = kwargs.pop("ekus", None)
+        self._key_usage = kwargs.pop("key_usage", None)
+        self._content_type = kwargs.pop("content_type", None)
+        self._validity_in_months = kwargs.pop("validity_in_months", None)
+        self._lifetime_actions = kwargs.pop("lifetime_actions", None)
+        self._certificate_type = kwargs.pop("certificate_type", None)
+        self._certificate_transparency = kwargs.pop("certificate_transparency", None)
+        self._san_emails = kwargs.pop("san_emails", None) or None
+        self._san_dns_names = kwargs.pop("san_dns_names", None) or None
+        self._san_upns = kwargs.pop("san_upns", None) or None
+
+        if not (self._san_emails or not self._san_upns or not self._san_dns_names or not self._subject_name):
```

Comment:
This expression is true only when we get everything except `san_emails`. not a and not b = not (a or b) != not (a or not b)

Reply:
totally missed that, thanks so much!